code/game/objects/items/devices/inhaler.dm

```diff
@@ -11,10 +11,26 @@
 	throw_range = 20
 	force = 0
 	var/cooldown = 0
-	
+
 /obj/item/device/inhaler/attack_self(mob/user)
 	if(cooldown < world.time - 8  && ishuman(user))
-		var/mob/living/carbon/human/H = user
-		playsound(user, 'sound/effects/spray2.ogg', 20, 1)
-		H.reagents.add_reagent(ALBUTEROL, 5)
-		cooldown = world.time
+		puff(user)
+
+/obj/item/device/inhaler/attack(mob/living/M, mob/user)
+	if(!ishuman(M) || user.zone_sel.selecting != "mouth" || user.a_intent != I_HELP)
+		return ..()
+	if(cooldown < world.time - 8)
+		var/mob/living/carbon/human/H = M
+		var/obj/item/mouth_protection = H.get_body_part_coverage(MOUTH)
+		if(mouth_protection)
+			to_chat(user, "<span class='warning'>Remove their \the [mouth_protection] first!</span>")
+			return 1
```

Comment:
Why returning `1` (true) if it's a falling case ? Might be mistaken here

Reply:
From what I understand, returning 1 means it terminates the click code so it doesn't do anything else. Not entirely sure if it means anything if it's attack and not preattack though. Just imitating how the medical stacks have it to be honest.

Reply:
That's good enough for me